cmd/crictl/exec.go

```diff
@@ -160,11 +168,35 @@ func Exec(ctx context.Context, client internalapi.RuntimeService, opts execOptio
 	}
 
 	logrus.Debugf("Exec URL: %v", URL)
-	return stream(ctx, opts.stdin, opts.tty, URL)
+	return stream(ctx, opts.stdin, opts.tty, opts.transport, URL)
+}
+
+func getExecutor(protocol string, url *url.URL) (exec remoteclient.Executor, err error) {
+	config := &restclient.Config{TLSClientConfig: restclient.TLSClientConfig{Insecure: true}}
+
+	switch protocol {
+	case "spdy":
+		exec, err = remoteclient.NewSPDYExecutor(config, "POST", url)
+	case "websocket":
+		exec, err = remoteclient.NewWebSocketExecutor(config, "GET", url.String())
```

Comment:
Can we use `remoteclient.NewWebSocketExecutorForProtocols` and make the protocols CLI configurable?

Reply:
ofcourse, default use StreamProtocolV5Name?

Reply:
@duguhaotian yeah

Reply:
change option to `crictl --websocket="x..channel.k8s.io"`, if not set `--websocket`, will use spdy

Reply:
> @duguhaotian yeah, this way we'll not require to flip the default again in the future once [kubernetes/kubernetes#122263](https://github.com/kubernetes/kubernetes/issues/122263) got resolved.

fixed